x/clerk/side_handler_test.go

```diff
@@ -0,0 +1,279 @@
+	// 	// event := &statesender.StatesenderStateSynced{
+	// 	// 	Id:              new(big.Int).SetUint64(msg.Id),
+	// 	// 	ContractAddress: common.BytesToAddress(msg.ContractAddress.Bytes()),
+	// 	// 	Data:            msg.Data,
+	// 	// }
+	// 	// suite.contractCaller.On("DecodeStateSyncedEvent", chainParams.ChainParams.StateSenderAddress, txReceipt, logIndex).Return(event, nil)
+
+	// 	// execute handler
+	// 	result := suite.sideHandler(ctx, &msg)
+	// 	// require.Equal(t, uint32(sdk.CodeOK), result.Code, "Side tx handler should be success")
+	// 	require.Equal(t, tmprototypes.SideTxResultType_YES, result.Result, "Result should be `yes`")
+
+	// 	// there should be no stored event record
+	// 	storedEventRecord, err := app.ClerkKeeper.GetEventRecord(ctx, id)
+	// 	require.Nil(t, storedEventRecord)
+	// 	require.Error(t, err)
+	// })
+
+	// t.Run("NoReceipt", func(t *testing.T) {
+	// 	suite.contractCaller = mocks.IContractCaller{}
+
+	// 	logIndex := uint64(200)
+	// 	blockNumber := uint64(51)
+	// 	txHash := hmCommon.HexToHeimdallHash("no receipt hash")
+
+	// 	msg := types.NewMsgEventRecord(
+	// 		addr1,
+	// 		txHash,
+	// 		logIndex,
+	// 		blockNumber,
+	// 		id,
+	// 		addr1,
+	// 		make([]byte, 0),
+	// 		suite.chainID,
+	// 	)
+
+	// 	// mock external calls -- no receipt
+	// 	suite.contractCaller.On("GetConfirmedTxReceipt", txHash.EthHash(), chainParams.MainchainTxConfirmations).Return(nil, nil)
+	// 	suite.contractCaller.On("DecodeStateSyncedEvent", chainParams.ChainParams.StateSenderAddress, nil, logIndex).Return(nil, nil)
+
+	// 	// execute handler
+	// 	result := suite.sideHandler(ctx, &msg)
+	// 	fmt.Printf("result %+v\n", result)
+	// 	// require.NotEqual(t, uint32(sdk.CodeOK), result.Code, "Side tx handler should fail")
+	// 	// require.Equal(t, tmprototypes.SideTxResultType_YES, result.Result, "Result should be `skip`")
+	// })
+
+	// t.Run("NoLog", func(t *testing.T) {
+	// 	suite.contractCaller = mocks.IContractCaller{}
+
+	// 	logIndex := uint64(100)
+	// 	blockNumber := uint64(510)
+	// 	txReceipt := &ethTypes.Receipt{
+	// 		BlockNumber: new(big.Int).SetUint64(blockNumber),
+	// 	}
+	// 	txHash := hmCommon.HexToHeimdallHash("no log hash")
+
+	// 	msg := types.NewMsgEventRecord(
+	// 		addr1,
+	// 		txHash,
+	// 		logIndex,
+	// 		blockNumber,
+	// 		id,
+	// 		addr1,
+	// 		make([]byte, 0),
+	// 		suite.chainID,
+	// 	)
+
+	// 	// mock external calls -- no receipt
+	// 	suite.contractCaller.On("GetConfirmedTxReceipt", txHash.EthHash(), chainParams.MainchainTxConfirmations).Return(txReceipt, nil)
+	// 	suite.contractCaller.On("DecodeStateSyncedEvent", chainParams.ChainParams.StateSenderAddress, txReceipt, logIndex).Return(nil, nil)
+
+	// 	// execute handler
+	// 	result := suite.sideHandler(ctx, &msg)
+	// 	// require.NotEqual(t, uint32(sdk.CodeOK), result.Code, "Side tx handler should fail")
+	// 	require.Equal(t, tmprototypes.SideTxResultType_YES, result.Result, "Result should be `skip`")
+	// })
+}
```

Comment:
Any dependency on other modules?

Reply:
based on chain params which is not working
needed @murthyvitwit help